### DISCUSSION TITLE
Conditionally show live output from the Helm binary

### DISCUSSION THREAD

**Discussion by rodrigorfk** [helmfile/helmfile#283] [Ideas]
Hi there, first of all, thank you very much for the great tool and project you are providing to the community.

I would like to bring into attention again what was previously discussed in the following issues/threads:
- https://github.com/roboll/helmfile/issues/2054
- https://github.com/roboll/helmfile/issues/1320#issuecomment-986485490

I have a requirement to provide tracking for resources during a helm upgrade, the ultimate goal is to enhance developer experience during a helm release, enabling instant feedback of what is happening underneath the release when using the `wait` and `atomic` capabilities of Helm.
With that being said, [Werf](https://github.com/werf/werf) provides a compatible Helm CLI interface when running with the `WERF_HELM3_MODE=1` environment variable exported, we can then point the `--helm-binary/-b` Helmfile global flag to the Werf CLI and Werf will provide [resource tracking](https://werf.io/documentation/v1.2/advanced/helm/deploy_process/steps.html#resource-tracking).
The only missing part of the puzzle from Helmfile is that Helmfile will only show the output of the Helm binary at the end of the release, suppressing any live output generated during it's execution, breaking the beauty of the live resource tracking during the release.

My suggestion here is to added another Helmfile global flag, let's call it `--show-live-output`, which will be disabled by default, to maintain the current behaviour, and once the user opt for it, the Output func of the runner.go [here](https://github.com/helmfile/helmfile/blob/86bfb5689abe535ea42ba4d51b2622304e332375/pkg/helmexec/runner.go) will pipe the stdout/stderr of the Helm binary live to the stdout/stderr of Helmfile itself, stop buffering it and waiting to write it only once the command is finished.

For your information I have a running patched version of Helmfile that does exactly that, I'm keen to open a pull request and contribute with it, I just want to check before if that is something you believe will be useful to incorporate into to project or if you have a better idea on how to solve it.

Thanks in advance.

**Comment by yxxhero** [comment]
@rodrigorfk PR is welocme.

**Reply by rodrigorfk** [reply]
Thanks @yxxhero , I have submitted the following PR https://github.com/helmfile/helmfile/pull/286, please let me know if having only this discussion here will be enough to move forward, or if I shall open a normal issue. Thanks.
